ledger/src/shred.rs

```diff
@@ -570,7 +591,12 @@ impl Shred {
     }
 
     // Returns the portion of the shred's payload which is erasure coded.
-    pub(crate) fn erasure_block(self) -> Vec<u8> {
+    pub(crate) fn erasure_block(self) -> Result<Vec<u8>, Error> {
+        if self.payload.len() != SHRED_PAYLOAD_SIZE {
+            return Err(Error::InvalidPayloadSize {
+                size: self.payload.len(),
+            });
+        }
```

Comment:
nit: Could consolidate the 3 instances of this down to one common helper function: